sw/device/silicon_creator/rom/rom.c

```diff
@@ -706,17 +712,26 @@ static rom_error_t rom_boot(const manifest_t *manifest, uint32_t flash_exec) {
     otp_read(OTP_CTRL_PARAM_CREATOR_SW_CFG_IMMUTABLE_ROM_EXT_SHA256_HASH_OFFSET,
              immutable_rom_ext_hash.digest, kHmacDigestNumWords);
     for (size_t i = 0; i < kHmacDigestNumWords; ++i) {
-      HARDENED_CHECK_EQ(immutable_rom_ext_hash.digest[i],
-                        actual_immutable_section_digest.digest[i]);
+      if (immutable_rom_ext_hash.digest[i] !=
+          actual_immutable_section_digest.digest[i]) {
+        verify_result = kErrorRomImmSection;
+      }
+    }
+    // If address translation is enabled, adjust the entry_point.
+    if (launder32(manifest->address_translation) == kHardenedBoolTrue) {
```

Comment:
Just to confirm: this adjustment of the immutable section entry point based on the address translation enablement setting being moved to after the hashing of the immutable section is a nop right? I.e. moving of this block (on 721--727) from (682--689) should have no impact?

Reply:
No - its required.  The addresses to hash must be mapped.  Since we're now performing the imm_section check before `rom_boot` (which sets up the remap window and ePMP), we have to perform the digest over our view of the section in flash.

After we've checked the digest, we can then adjust the entry address for translation.  We won't attempt to access via the translated address (in `rom_boot` until after the remap window and ePMP are configured).